lmdeploy/messages.py

```diff
@@ -116,6 +120,10 @@ def __post_init__(self):
         assert self.top_p > 0 and self.top_p <= 1  # (0, 1]
         assert self.top_k >= 0, 'top_k can not be a negative integer'
         assert self.temperature >= 0 and self.temperature <= 2  # [0,2]
+        if self.do_sample is False:
+            self.top_k = 1
+            self.temperature = 1.0
```

Comment:
should we set `temperature` 0?

Reply:
Top_k is enough. Setting temperature and repetition_penalty to default value is to skip unnecessary operation.